diff --git a/node_modules/react-joyride/es/components/Portal.js b/node_modules/react-joyride/es/components/Portal.js
index 9980540..a259c03 100644
--- a/node_modules/react-joyride/es/components/Portal.js
+++ b/node_modules/react-joyride/es/components/Portal.js
@@ -1,5 +1,5 @@
 import PropTypes from 'prop-types';
-import ReactDOM from 'react-dom';
+import { createRoot } from 'react-dom/client';
 import { Component } from 'react';
 
 export default class Portal extends Component {
@@ -11,6 +11,7 @@
 
   componentDidMount() {
     this.renderPortal();
+    this.root = createRoot(this.node);
   }
 
   componentDidUpdate() {
@@ -21,7 +22,7 @@
       return;
     }
 
-    ReactDOM.unmountComponentAtNode(this.node);
+    this.root.unmount();
 
     if (this.node.parentNode) {
       this.node.parentNode.removeChild(this.node);
@@ -32,7 +33,7 @@
     if (!this.props.portal) {
       return;
     }
-    ReactDOM.render(this.props.portal, this.node);
+    this.root.render(this.props.portal);
   }
 
   render() {
diff --git a/node_modules/react-joyride/lib/components/Portal.js b/node_modules/react-joyride/lib/components/Portal.js
index 45145f3..a43411a 100644
--- a/node_modules/react-joyride/lib/components/Portal.js
+++ b/node_modules/react-joyride/lib/components/Portal.js
@@ -11,7 +11,7 @@
 var _propTypes = _interopRequireDefault(require("prop-types"));
 
 var _react = require("react");
-var _reactDom = _interopRequireDefault(require("react-dom"));
+var _client = require("react-dom/client");
 
 function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
 
@@ -32,6 +32,7 @@
 
     _this.renderPortal = function () {
       if (!_this.props.portal) {
         return;
       }
@@ -39,7 +40,7 @@
-      _reactDom.default.render(_this.props.portal, _this.node);
+      _this.root.render(_this.props.portal);
     };
 
     _this.node = document.createElement('div');
@@ -51,6 +52,7 @@
 
   _createClass(Portal, [{
     key: "componentDidMount",
     value: function componentDidMount() {
       this.renderPortal();
+      this.root = (0, _client.createRoot)(this.node);
     }
   }, {
     key: "componentDidUpdate",
@@ -62,7 +64,7 @@
         return;
       }
 
-      _reactDom.default.unmountComponentAtNode(this.node);
+      this.root.unmount();
 
       if (this.node.parentNode) {
         this.node.parentNode.removeChild(this.node);
