diff --git a/node_modules/react-joyride/dist/index.mjs b/node_modules/react-joyride/dist/index.mjs
index 81b6817..5164472 100644
--- a/node_modules/react-joyride/dist/index.mjs
+++ b/node_modules/react-joyride/dist/index.mjs
@@ -1,6 +1,6 @@
 import React, { useMemo, useCallback, useState, useEffect, useRef, forwardRef, createContext, useContext } from 'react';
 import { styled } from 'goober';
-import { createPortal, findDOMNode, unmountComponentAtNode as unmountComponentAtNode$1 } from 'react-dom';
+import { createPortal, findDOMNode } from 'react-dom';
 import { createRoot } from 'react-dom/client';
 import { computePosition, offset, flip, shift, autoPlacement, arrow as arrow$1, hide } from '@floating-ui/dom';
 import { deepmerge } from 'deepmerge-ts';
@@ -207,7 +207,9 @@
         if (isReact18) {
           root.current.unmount();
         } else {
-          unmountComponentAtNode$1(portal.current);
+          import('react-dom').then(function (ReactDOM) {
+            ReactDOM.unmountComponentAtNode(portal.current);
+          });
         }
         document.body.removeChild(portal.current);
       }
